pyface/data_view/data_formats.py

```diff
@@ -0,0 +1,293 @@
+# (C) Copyright 2005-2020 Enthought, Inc., Austin, TX
+# All rights reserved.
+#
+# This software is provided without warranty under the terms of the BSD
+# license included in LICENSE.txt and may be redistributed only under
+# the conditions described in the aforementioned license. The license
+# is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+# Thanks for using Enthought open source!
+import csv
+from functools import partial
+from io import BytesIO, StringIO
+import os
+import json
+
+from .i_data_wrapper import DataFormat, text_format
+
+
+# Scalar formats
+
+def to_json(data, default=None):
+    """ Serialize an object to a JSON bytestring.
+
+    Parameters
+    ----------
+    data : any
+        The data to be serialized.
+    default : callable or None
+        Callable that takes a Python object and returns a JSON-serializable
+        data structure.
+
+    Returns
+    -------
+    raw_data : bytes
+        The serialized data as a bytestring.
+    """
+    str_data = json.dumps(data, default=default, separators=(',', ':'))
+    return str_data.encode('utf-8')
+
+
+def from_json(raw_data, object_hook=None):
+    """ Deserialize a JSON bytestring.
+
+    Parameters
+    ----------
+    raw_data : bytes
+        The serialized JSON data as a byte string.
+    object_hook : callable
+        Callable that takes a dictionary and returns an corresponding
+        Python object.
+
+    Returns
+    -------
+    data : any
+        The data extracted.
+    """
+    return json.loads(raw_data.decode('utf-8'), object_hook=object_hook)
+
+
+#: The text/plain format with utf-8 encoding.
+standard_text_format = text_format()
+
+#: The text/html format with utf-8 encoding.
+html_format = text_format(mimetype='text/html')
+
+#: A generic JSON format.
+json_format = DataFormat('application/json', to_json, from_json)
+
+
+# 1D formats
+
+def to_csv_row(data, delimiter=',', encoding='utf-8', **kwargs):
+    """ Serialize a list to a single-row CSV bytestring.
+
+    Parameters
+    ----------
+    data : list
+        The list to be serialized.  Any elements which are not strings will
+        be converted to strings by calling ``str()``.
+    delimiter : str
+        The CSV delimiter.
+    encoding : str
+        The encoding of the bytes
+    **kwargs
+        Additional arguments to csv.writer.
+
+    Returns
+    -------
+    raw_data : bytes
+        The serialized data as a bytestring.
+    """
+    with StringIO() as fp:
+        writer = csv.writer(fp, delimiter=delimiter, **kwargs)
+        writer.writerow(data)
+        return fp.getvalue().encode(encoding)
```

Comment:
As with any context manager when you leave the block, it will be in the variable `fp`.  It will have been `closed`, however.

Not sure why I used a context manager here - it may have simply been because of code evolution from an earlier iteration of the API where I was passing in a file-like.